### PR TITLE
Voting System Improvements and Fixes

### PR DESCRIPTION
**Pull request by Krausus**
Initially, I only intended to add full results output to custom votes, but then things got out of hand...

- Makes custom votes show a sorted list of votes for each option upon completion, and others show total votes for the winning option.
  - dreamseeker_2016-07-23_01-24-22
  - dreamseeker_2016-07-23_01-30-25
- Makes the voting panel automatically update every second.
  - This used to be a feature of the voting panel, but it was done by constantly re-creating the window, which didn't work very well.
  - Now, it sends javascript-based updates, so excess updates will simply be ignored.
    - Also, updating the current vote results won't be all *clicky.*
- Makes the voting panel use the browser datum (pictured below).
  - In addition to looking nice, this has convenient support for detecting when a window closes, so we can stop sending vote updates.
- Turns the voting notice into a clickable link.
  - Players no longer have to type in or hunt for the `vote` verb.
- Fixes admin status not properly being checked when the voting panel is refreshed.
  - This is why the "cancel vote" link would almost immediately disappear from an active vote.
- Replaces vote process with a spawned loop in the vote controller.
  - We had a process dedicated to ticking this one, single datum, every second. The process didn't even start until the round did, despite votes being usable before round start.
- Removes vote processing from the pre-game game ticker.
  - I assume this was a sloppy work-around for the vote process not running during the pre-game. It was ticking the vote controller *so fast*, though.
- Removes relative pathing from vote controller, along with other minor refactoring.
- Adds logging of (player-made) votes being started, and logging+notification of votes being cancelled by an admin.
- Adds the vote controller to the "debug controller" list.

The voting panel now looks like this:

:cl:
tweak: The voting panel now uses the browser datum, which means it looks nicer.
tweak: The voting panel now updates while a vote is in progress.
tweak: When a vote is started, it will now include a link to open the vote panel, as an alternative to typing in "vote".
tweak: Custom votes will now show full voting results upon completion.
bugfix: Admins should no longer lose the "cancel vote" link in the voting panel.
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
Time to pretend this is a democracy.

**Comment by Kyep** [comment]
Great!

Is it still the case that only heads (+seniors?) can create votes?

File: code/controllers/voting.dm [hunk 1]
```diff
@@ -14,345 +14,386 @@ datum/controller/vote
...
+		if(current_votes[C.ckey] == i)
+			. += "<li><b><a href='?src=\ref[src];vote=[i]'>[choices[i]] ([votes] vote\s)</a></b></li>"
+		else
+			. += "<li><a href='?src=\ref[src];vote=[i]'>[choices[i]] ([votes] vote\s)</a></li>"
+
+	. += "</ul>"
+
+
+/datum/controller/vote/Topic(href,href_list[],hsrc)
+	if(!usr || !usr.client)	return	//not necessary but meh...just in-case somebody does something stupid
```
Comment (tigercat2000) [inline code/controllers/voting.dm]: newline returrrrn

**Comment by tigercat2000** [comment]
> Is it still the case that only heads (+seniors?) can create votes?

It's never been the case that only heads can create votes. Anyone with R_ADMIN can do it.

File: code/controllers/voting.dm (hunk 1 above)
Comment (Krausus) [inline code/controllers/voting.dm]: Oh god. Please don't nitpick all the code I haven't actually touched. I made a separate commit just for the code I actually changed, and even fixed some of those.

Comment (tigercat2000) [inline code/controllers/voting.dm]: rawr

**Comment by monster860** [comment]
The Research Director has created a new vote:

```Toggle bomb cap off?

Yes
No
```

File: code/controllers/voting.dm [hunk 2]
```diff
@@ -20,9 +20,13 @@ var/global/list/round_voters = list() //Keeps track of the individuals voting fo
 			qdel(vote)
 		vote = src
 	spawn(0)
-		while(TRUE)
-			process()
-			sleep(10)
+		while(!gcDestroyed)
+			try
```
Comment (PJB3005) [inline code/controllers/voting.dm]: I don't see the point of this.

It ruins stack traces and the stack trace tells you that it's in the voting controller anyways.

Comment (Krausus) [inline code/controllers/voting.dm]: The point of try/catch is that you don't need extra code to handle detecting and recovering from runtimes, which makes the code cleaner. Losing the stack trace is the only downside, but most of the time, it's simple enough to just manually trace back through the calls. Especially in a situation like this, with very few calls out of the vote controller code.

**Pull request merged**